recipes/libtins/all/conanfile.py

```diff
@@ -29,106 +40,92 @@ class LibTinsConan(ConanFile):
         "with_dot11": True,
     }
 
-    exports_sources = ["CMakeLists.txt"]
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
     @property
-    def _build_subfolder(self):
-        return "build_subfolder"
+    def _min_cppstd(self):
+        return 11
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("libpcap/1.10.1")
+        self.requires("libpcap/1.10.1", transitive_headers=True, transitive_libs=True)
         if self.options.with_ack_tracker:
-            self.requires("boost/1.79.0")
+            self.requires("boost/1.81.0", transitive_headers=True, transitive_libs=True)
         if self.options.with_wpa2:
-            self.requires("openssl/1.1.1q")
+            self.requires("openssl/3.1.0")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["LIBTINS_BUILD_EXAMPLES"] = False
+        tc.variables["LIBTINS_BUILD_TESTS"] = False
+        tc.variables["LIBTINS_BUILD_SHARED"] = self.options.shared
+        tc.variables["LIBTINS_ENABLE_CXX11"] = True
+        tc.variables["LIBTINS_ENABLE_ACK_TRACKER"] = self.options.with_ack_tracker
+        tc.variables["LIBTINS_ENABLE_WPA2"] = self.options.with_wpa2
+        tc.variables["LIBTINS_ENABLE_DOT11"] = self.options.with_dot11
+        tc.generate()
+        deps = CMakeDeps(self)
```

Comment:
```suggestion
        deps = CMakeDeps(self)
        deps.set_property("libpcap", "cmake_file_name", "PCAP")
```

There is a better trick than patching the upstream CMake, you can override the expected cmake target and file name

https://docs.conan.io/2/reference/tools/cmake/cmakedeps.html#overwrite-properties-from-the-consumer-side-using-cmakedeps-set-property

You still need to patch the target, because Conan will generate `PCAP_LIBRARIES` not `PCAP_LIBRARY`.

Reply:
That's good to know, thanks!